src/main/java/seedu/address/logic/commands/DeleteRestaurantNoteCommand.java

```diff
@@ -0,0 +1,154 @@
+     * @param lineBad number of a specific note in the bad food information stored
+     */
+    public DeleteRestaurantNoteCommand(Index index, ArrayList<Integer> lineRec, ArrayList<Integer> lineGood,
+                                       ArrayList<Integer> lineBad) {
+        requireAllNonNull(index, lineRec, lineGood, lineBad);
+
+        this.index = index;
+        this.lineRec = lineRec;
+        this.lineGood = lineGood;
+        this.lineBad = lineBad;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Restaurant> lastShownList = model.getFilteredRestaurantList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_RESTAURANT_DISPLAYED_INDEX);
+        }
+
+        Restaurant restaurantToEdit = lastShownList.get(index.getZeroBased());
+        ArrayList<Note> recNotes = new ArrayList<>();
+        ArrayList<Note> goodNotes = new ArrayList<>();
+        ArrayList<Note> badNotes = new ArrayList<>();
+
+        if (lineRec.size() != 0) {
+            if (restaurantToEdit.getRecommendedFood().size() == 0) {
+                throw new CommandException(Messages.MESSAGE_NO_RECOMMENDED_FOOD);
+            }
+            for (Integer i : lineRec) {
+                if (i > restaurantToEdit.getRecommendedFood().size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_LINE_NUMBER_REC);
+                }
+                recNotes.add(restaurantToEdit.getRecommendedFood().get(i - 1));
+            }
+        }
+
+        if (lineGood.size() != 0) {
+            if (restaurantToEdit.getGoodFood().size() == 0) {
+                throw new CommandException(Messages.MESSAGE_NO_GOOD_FOOD);
+            }
+            for (Integer i : lineGood) {
+                if (i > restaurantToEdit.getGoodFood().size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_LINE_NUMBER_GOOD);
+                }
+                goodNotes.add(restaurantToEdit.getGoodFood().get(i - 1));
+            }
+        }
+
+        if (lineBad.size() != 0) {
+            if (restaurantToEdit.getBadFood().size() == 0) {
+                throw new CommandException(Messages.MESSAGE_NO_BAD_FOOD);
+            }
+            for (Integer i : lineBad) {
+                if (i > restaurantToEdit.getBadFood().size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_LINE_NUMBER_BAD);
+                }
+                badNotes.add(restaurantToEdit.getBadFood().get(i - 1));
+            }
+        }
+
+        for (Note n : recNotes) {
+            restaurantToEdit.getRecommendedFood().remove(n);
+        }
+        for (Note n : goodNotes) {
+            restaurantToEdit.getGoodFood().remove(n);
+        }
+        for (Note n : badNotes) {
+            restaurantToEdit.getBadFood().remove(n);
+        }
```

Comment:
Why you do this separately arh and not in the above checks?

Reply:
If one is valid, and the other is invalid, it might remove the first one before throwing an exception for the next